src/io/flutter/run/FlutterAppState.java

```diff
@@ -79,16 +83,15 @@ protected ProcessHandler startProcess() throws ExecutionException {
     ConnectedDevice device = null;
 
     // Only pass the current device in if we are showing the device selector.
-    final FlutterSettings settings = FlutterSettings.getInstance(project);
-    if (settings.isShowDevices()) {
+    if (service.isActive()) {
```

Comment:
I changed it to not be null (and added a `@NotNull` annotation to its definition).